front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts

```diff
@@ -240,7 +241,16 @@ export class CreateF3XStep1Component implements OnInit, OnDestroy {
         // Save report to Cash On Hand in the store if necessary by pulling the reports table data.
         this.reportService.getTableData().subscribe();
 
-        this.router.navigateByUrl(`/reports/f3x/create/step2/${report.id}`);
+        this.store
```

Comment:
Nested subscriptions should be avoided (like nested callbacks or promises https://angular-checklist.io/default/checklist/rxjs/Z1eFwa9).
We can flatten this by changing 239 to:
```
createSummary$ = this.f3xSummaryService.create(summary, this.formProperties);
selectCashOnHand$ = this.store.select(selectCashOnHand);
createSummary$.pipe(combineLatestWith(selectCashOnHand$)).subscribe(([report, cashOnHand]) => {
```
combineLatestWith https://rxjs.dev/api/operators/combineLatestWith

then, at line 244 you can just write your condition without setting up another subscription

Reply:
That's a super interesting idea.  I'll take a look.

Reply:
Yes, agree with Todd's comment about not nesting subscribes. @Elaine-Krauss-TCG : Also take a look at mergeMap() which I have found to be a super useful pattern to know as well: https://www.learnrxjs.io/learn-rxjs/operators/transformation/mergemap